app/models/ext_management_system.rb

```diff
@@ -71,9 +71,9 @@ def self.supported_types_and_descriptions_hash
   def hostname_uniqueness_valid?
     return unless hostname_required?
     return unless hostname.present? # Presence is checked elsewhere
+    # check uniqueness per provider type
 
-    existing_hostnames = Endpoint.where.not(:resource_id => id).pluck(:hostname).compact.map(&:downcase)
-
+    existing_hostnames = (self.class.all - [self]).map(&:hostname).compact.map(&:downcase)
     errors.add(:hostname, "has already been taken") if existing_hostnames.include?(hostname.downcase)
```

Comment:
@durandom I think I saw that the error was already being I18N'd.  I'll have dig up where I saw that.  Basically, the UI is taking the error message from the model and doing an I18N lookup from what I saw.